src/threads.js

```diff
@@ -3843,8 +3843,11 @@ Process.prototype.reportCombine = function (list, reporter) {
                 this.returnValueToParentContext(
                     list.length() ?
                         list.at(1)
-                        : (reporter.expression.selector === 'reportJoinWords' ?
-                            ''
+                        : (reporter.expression.selector === 'reportJoinWords' ? ''
+                            : reporter.expression.selector === 'reportVariadicAnd' ? true
+                            : reporter.expression.selector === 'reportVariadicOr' ? false
+                            : reporter.expression.selector === 'reportConcatenatedLists' ? new List()
+                            : reporter.expression.selector === 'reportCrossproduct' ? new List([new List()])
```

Comment:
I haven't tried this, but I don't think this is very readable

```suggestion
           initial_combine_value_for_selector(reporter.expression.selector)
```

```js
let initial_combine_value_for_selector = (selector) => {
  switch(selector) {
  case 'reportJoinWords':
    return '';
  case '':
     return ...
  default:
    return 0;
};
```

Reply:
Yes, now that there are five cases extracting them into a switch/case probably makes more sense. I'll work on that in a few days when I get a chance.